platform/mellanox/mlnx-platform-api/tests/test_chassis.py

```diff
@@ -372,3 +372,34 @@ def test_parse_dmi(self):
         chassis = Chassis()
         content = chassis._parse_dmi(os.path.join(test_path, 'dmi_file'))
         assert content.get('Version') == 'A4'
+    
+    def test_smartswitch(self):
+        DeviceDataManager.get_dpu_count = mock.MagicMock(return_value=4)
```

Comment:
Please extend the tests. These are unit tests we should not check only the positive scenarios but also negative ones. For each new API you've added, there should be at least one positive and several negative test cases. Please think and try to find potential corner cases and cover them with tests.

Reply:
Done